qdrant-landing/content/articles/dedicated-service.md

```diff
@@ -0,0 +1,144 @@
+date: 2023-10-01T10:00:00+03:00
+draft: false
+keywords:
+    - system architecture
+    - vector search
+    - best practices
+    - anti-patterns
+---
+
+
+Ever since the general public discovered that vector search significantly improves LLM answers,
+various vendors and enthusiasts have been arguing over where to properly store embeddings.
+
+Some says, that it's better to store them in a specialized search engines(aka vector database), others say that it's enough to use plugins for existing databases.
+
+Here are [just](https://nextword.substack.com/p/vector-database-is-not-a-separate) a [few](https://stackoverflow.blog/2023/09/20/do-you-need-a-specialized-vector-database-to-implement-vector-search-well/) of [them](https://www.singlestore.com/blog/why-your-vector-database-should-not-be-a-vector-database/).
+
+
+In this article, we are going to present our vision and arguments on the topic.
+We will try to explain why and when you acutally need a dedicated solution, debunk some ungrounded claims and anti-patterns that you should avoid while building a vector search system.
+
+We will analyze those theses of the Universal-Database-For-Everything advocates and discuss some of the best practices we recommend to adopt in real production-grade systems. 
+
+A table of contents:
+
+* *Each database vector will sooner or later introduce vector capabilities...* [[click](#each-database-vector-will-sooner-or-later-introduce-vector-capabilities-that-will-make-every-database-a-vector-database)]
+* *Having a dedicated vector database requires duplication of data.* [[click](#having-a-dedicated-vector-database-requires-duplication-of-data)]
+* *Having a dedicated vector database requires complex data synchronization.* [[click](#having-a-dedicated-vector-database-requires-complex-data-synchronization)]
+* *You have to pay for their uptime and data transfer.* [[click](#you-have-to-pay-for-their-uptime-and-data-transfer-of-both-solution)]
+* *What is more seamless than your current database adding vector search capability?* [[click](#what-is-more-seamless-than-your-current-database-adding-vector-search-capability)]
+* *Databases can support RAG use-case end-to-end.* [[click](#databases-can-support-rag-use-case-end-to-end)]
+
+
+## Responding to claims
+
+###### Each database vector will sooner or later introduce vector capabilities. That will make every database a Vector Database.
+
+
+The origins of this misconception lie in the careless use of the term Vector *Database*.
+When we think of a *database*, we subconsciously envision a relational database like Postgres or MySQL.
+Or more scientifically, a service built on ACID principles, that procivides transactions, strong consistency guarantees and atomicity.
+
+Majority of Vector Database are not *databases* in this sense.
+It would be more accurate to call them *search engines*, but unfortunately, the term marketing term *database* is already stuck and it is unlikely to change.
```

Comment:
```suggestion
It would be more accurate to call them *search engines*, but unfortunately, the marketing term *database* is already stuck and it is unlikely to change.
```